apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -289,267 +362,306 @@ export function buildMachine(
       events: {} as PaperHandlerStatusEvent,
     },
     id: 'bmd',
-    initial: 'not_accepting_paper',
+    initial: 'voting_flow',
     context: initialContext,
     on: {
-      PAPER_JAM: 'jammed',
-      JAMMED_STATUS_NO_PAPER: 'jam_physically_cleared',
-    },
-    states: {
-      // Initial state. Doesn't accept paper and transitions away when the frontend says it's ready to accept
-      not_accepting_paper: {
-        invoke: pollPaperStatus(),
-        on: {
-          // Paper may be inside the machine from previous testing or machine failure. We should eject
-          // the paper (not to ballot bin) because we don't know whether the page has been printed.
-          PAPER_INSIDE_NO_JAM: 'eject_to_front',
-          PAPER_PARKED: 'eject_to_front',
-          BEGIN_ACCEPTING_PAPER: 'accepting_paper',
-        },
+      PAPER_JAM: 'voting_flow.jammed',
+      JAMMED_STATUS_NO_PAPER: 'voting_flow.jam_physically_cleared',
+      PAT_DEVICE_CONNECTED: {
+        actions: assign({
+          patDevice: (_, event) => event.patDevice,
+        }),
+        target: 'pat_device_connected',
       },
-      accepting_paper: {
-        invoke: pollPaperStatus(),
-        on: {
-          PAPER_READY_TO_LOAD: 'loading_paper',
-        },
+      PAT_DEVICE_DISCONNECTED: {
+        actions: assign({
+          patDevice: () => undefined,
+        }),
+        // Without this target, the PAT device observable won't have an updated value
+        // for context.patDevice
+        target: 'voting_flow.history',
       },
-      loading_paper: {
-        invoke: [
-          pollPaperStatus(),
-          {
-            id: 'loadAndPark',
-            src: (context) => {
-              return loadAndParkPaper(context.driver);
+    },
+    invoke: [pollPatDeviceConnectionStatus()],
+    states: {
+      voting_flow: {
```

Comment:
Nesting the voting flow in its own state like this allows us to use a history node.

From docs:
> A history [state node](https://xstate.js.org/docs/guides/statenodes.html) is a special kind of state node that, when reached, tells the machine to go to the last state value of that region